app/screens/channel/channel_base.js

```diff
@@ -212,6 +215,10 @@ export default class ChannelBase extends PureComponent {
 
                 this.setState({channelsRequestFailed: false});
                 selectInitialChannel(teamId);
+
+                if (unreadChannelIds.length > 0) {
+                    preFetchUnreadChannels(unreadChannelIds);
```

Comment:
I think this makes more sense to call it from the `loadChannelsForTeam` action. One of the reasons for that is that every time `unreadChannelIds` changes this component will re-render and there is no real need for that.